MapboxNavigation/RouteMapViewController.swift

```diff
@@ -253,6 +260,61 @@ extension RouteMapViewController: NavigationMapViewDelegate {
         let route = routeController.routeProgress.route
         guard let coordinates = route.coordinates else  { return nil }
         
+        if let userLocation = mapView.userLocation,
+            let style = mapView.style,
+            recenterButton.isHidden {
+            
+            let streetsLanguages = ["zh", "ru", "fr", "es", "en"]
+            let roadLabelLayerIdentifier = "roadLabelLayer"
+            
+            let streetsSources = style.sources.flatMap {
+                $0 as? MGLVectorSource
+                }.filter {
+                    $0.isMapboxStreets
+            }
+            let streetsSourceIdentifiers = streetsSources.map {
+                $0.identifier
+            }
+            assert(!streetsSourceIdentifiers.isEmpty, "Style must contain the source `mapbox.mapbox-streets-v7`")
+            
+            if let mapboxSteetsSource = streetsSources.first, style.layer(withIdentifier: roadLabelLayerIdentifier) == nil{
+                let streetLabelLayer = MGLLineStyleLayer(identifier: roadLabelLayerIdentifier, source: mapboxSteetsSource)
+                streetLabelLayer.sourceLayerIdentifier = "road_label"
+                
+                // If the opacity is set to 0, the feature will be ignored in `mapView.visibleFeatures()`
+                streetLabelLayer.lineOpacity = MGLStyleValue(rawValue: 0.001)
+                streetLabelLayer.lineWidth = MGLStyleValue(rawValue: 10)
+                streetLabelLayer.lineColor = MGLStyleValue(rawValue: .white)
+                style.addLayer(streetLabelLayer)
+            }
+            
+            if let userPuck = mapView.view(for: userLocation) {
+                let features = mapView.visibleFeatures(in: userPuck.frame, styleLayerIdentifiers: Set([roadLabelLayerIdentifier]))
+                
+                for feature in features {
+                    
+                    var key = "name"
+                    if let language = Locale.preferredLanguages.first!.components(separatedBy: "-").first,
+                        streetsLanguages.contains(language) || language == "zh-Hans" {
+                        key += "_\(language)"
+                    }
+                    
+                    if let name = feature.attribute(forKey: key) as? String {
+                        if name != wayNameLabel.text {
+                            currentWayNameCounter += 1
+                        }
+                        print(currentWayNameCounter)
+                        if currentWayNameCounter >= 10 {
```

Comment:
It takes ten location updates (at least ten seconds) to update the label? What if the user has already turned onto another road? Wouldn’t they get confused when they glance at the label in the meantime and find that it’s still the old road name?

Instead of delaying the update, let’s check whether the feature matches the route line. If more than one consecutive coordinates in the feature’s `coordinates` are also in the route step’s `coordinates`, then use the feature for the label. Otherwise, ignore the feature.